src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnRunner.java

```diff
@@ -699,6 +697,22 @@ private SpawnResult handleError(
         .build();
   }
 
+  static List<Property> getExtraPlatformProperties(Spawn spawn, Path execRoot) {
+    // Hack on Windows. The included headers dumped by cl.exe in stdout contain absolute paths.
+    // When compiling the file from different workspace, the shared cache will cause header
+    // dependency checking to fail. This was initially fixed by a hack (see
+    // https://github.com/bazelbuild/bazel/issues/9172 for more details), but is broken again due
+    // to cl/356735700. We include workspace name here so action results from different workspaces
+    // on Windows won't be shared.
+    boolean isWindows = System.getProperty("os.name").startsWith("Windows");
+    if (isWindows) {
+      return ImmutableList
+          .of(Property.newBuilder().setName("workspace").setValue(execRoot.getBaseName()).build());
```

Comment:
Do I understand correctly that `workspace` is not interpreted by RBE and is just to differentiate the cache keys? In that case, call it something more descriptive, e.g. `BAZEL_WORKSPACE_DISAMBIGUATION_FOR_MSVC_HACK`.

Reply:
Exactly!